### ISSUE TITLE
Addition of the tag: tag_electricity_storage_types.yaml

### ISSUE THREAD

**Issue by erikfilias** [openENTRANCE/openentrance#183]
Some variables have common features. 
That is the case of:
```
- Maximum Storage|Electricity|Hydro|Reservoir:
- Maximum Storage|Electricity|Hydro|Pumped Storage:
- Maximum Storage|Electricity|Energy Storage System:
- Maximum Storage|Electricity|Energy Storage System|Lithium-Ion:
- Maximum Storage|Electricity|Energy Storage System|Redox Flow:
- Maximum Storage|Electricity|Energy Storage System|Compressed Air:
```
In this sense, I suggest the addition of the tag `tag_electricity_storage_types.yaml` to be concise and standardize all the possible characteristics of the storage because I have found that some storage does not have defined some of their characteristics.

@sandrinecharousset, @danielhuppmann, @arght, @loeffko, What do you think?

If you agree, I can deal with ordering and creating this tag.

**Comment by sandrinecharousset** [comment]
@erikfilias it is not exactly the case....
They do have some common variables but not so many and they do have a lot of different ones.....
Hydro|* do have Capacity , Variable Cost, Capital Cost....
Hydro|Pumped Storage has efficiency
The energy storage system variables do not have those features
They have Charge/discharge, roundtrip coefficient, levelysed costs 
So I would suggest not to do that.....

**Comment by sandrinecharousset** [comment]
@erikfilias I think we have different categories of variables
1/ those for assets in {electricity input} ; electricity input covers all assets that produce electricity, including Hydro but not energy storage system
2/ assets in {energy storage system} which covers assets which are storage but not producing electricity
3/ Hydro which is specific as it both produces electricity and stores energy

Regarding the variables specific to storage we have:
- maximum storage,  for Hydro|Reservoir, Hydro|Pumped Storage and Energy Storage System|*
- minimum storage  for Hydro|Reservoir, Hydro|Pumped Storage , not for ESS but could be
- maximum charge/ maxium discharge for Hydro|Pumped Storage and ESS , not for Reservoir but could be (even if Reservoir usually does not have pumping, but it could happen anyway)
- seasonality, for all

Then some variables where we need to decide sth:
Pumping / Turbine efficiency for Reservoir and Pumped Storage / Roundtrip coefficient for ESS => these could be merged . Seems Roundtrip efficiency=pumping afficiency * turbine efficnecy , so I would keep 3 variables but perhaps with a different name such as charge efficiency / discharge efficiency / roundtrip efficiency ? What do you think?

And finally we have Inflows for Reservoir and Pumped Storage. In plan4EU we also can use Inflows for ESS, but usually it is negative Inflows (it is named 'demand' in the model as it represents an external demand that the battery faces); We could maybe keep Inflows and add Demand for both Hydro and ESS?

Then there is this Maximum Energy Charge|Electricity|Energy Storage System:
    description: This is the upper bound that energy storage system could charge energy
      at time "t". I do not fully understand the difference with Maximum Charge (maximum charge is in GW , not GWh but this is equivalent, no? and I cannot fully understand how the charge at time t can be defined without ambiguity, given that the timestep duration would be missing....; I would delete this variable but you may need it

And there is 'Spillage' which for me should be Maximum Spillage (as in power-plant we are not listing the variables for describing the operation, but the variables which caracterize the power plants; and for me it should be in GW, not GWh, for the same reason as above

In summary, all the above variables could be summarized with a Electricity Storage tag, yes.

**Comment by erikfilias** [comment]
@sandrinecharousset, 
1/ the third variable related to the efficiency could be defined as you suggested: `charge efficiency` and `discharge efficiency`. I think it works well for our purposes.
2/ In openTEPES, we have 'inflows' and 'outflows'. The inflows are defined as yours and our outflows cover exactly the 'external demand' as you defined. In this sense, I would define the outflows for both hydro and ESS, but I think 'External Demand' should be more readable. Thus, I suggest this last one. What do you think?
3/ About the 'Maximum Charge'. Yeah, we need it since it is part of the main variables for ESS/PSH management. It is a parameter like 'Maximum Active Power' in 'MW', but 'Maximum Charge' is related to how much active power can charge the 'Storage Unit' as maximum. Perhaps, I should change the definition and the unit. What would you like to suggest about it?/ 4
4/ About the spillage. Yeah, I also agree. I can modify it. Do you think that all these changes should be in the same PR?

Thanks for the comment :)

**Comment by sandrinecharousset** [comment]
1/ the third variable related to the efficiency could be defined as you suggested: charge efficiency and discharge efficiency. I think it works well for our purposes.
	OK !

2/ In openTEPES, we have 'inflows' and 'outflows'. The inflows are defined as yours and our outflows cover exactly the 'external demand' as you defined. In this sense, I would define the outflows for both hydro and ESS, but I think 'External Demand' should be more readable. Thus, I suggest this last one. What do you think?
	I would keep Inflows and outflows, i think it is very easy to understand (Inflows come in, Outflows come out…..) ; We could add in the description of the outflows that it can be eg an ‘external demand’ (but for the hydro could also be many different things)

3/ About the 'Maximum Charge'. Yeah, we need it since it is part of the main variables for ESS/PSH management. It is a parameter like 'Maximum Active Power' in 'MW', but 'Maximum Charge' is related to how much active power can charge the 'Storage Unit' as maximum. Perhaps, I should change the definition and the unit. What would you like to suggest about it?
	We already have Maximum Charge and Maximum Discharge, unit GW or MW , but there is also Maximum Energy Charge which I find useless and difficult to understand as it deals with the duration of a timestep, which is not explicit in the dataformat….. My suggestion was to keep Maximum Charge and Maximum Discharge but to remove Maximum Energy Charge

4/ About the spillage. Yeah, I also agree. I can modify it. 
	Good
Do you think that all these changes should be in the same PR?
	Yes, won’t be too big…..

@erikfilias I think we can start the PR, right?

**Comment by erikfilias** [comment]
> 1/ the third variable related to the efficiency could be defined as you suggested: charge efficiency and discharge efficiency. I think it works well for our purposes.  OK !
> 
> 2/ In openTEPES, we have 'inflows' and 'outflows'. The inflows are defined as yours and our outflows cover exactly the 'external demand' as you defined. In this sense, I would define the outflows for both hydro and ESS, but I think 'External Demand' should be more readable. Thus, I suggest this last one. What do you think?  I would keep Inflows and outflows, i think it is very easy to understand (Inflows come in, Outflows come out…..) ; We could add in the description of the outflows that it can be eg an ‘external demand’ (but for the hydro could also be many different things)
> 
> 3/ About the 'Maximum Charge'. Yeah, we need it since it is part of the main variables for ESS/PSH management. It is a parameter like 'Maximum Active Power' in 'MW', but 'Maximum Charge' is related to how much active power can charge the 'Storage Unit' as maximum. Perhaps, I should change the definition and the unit. What would you like to suggest about it?  We already have Maximum Charge and Maximum Discharge, unit GW or MW , but there is also Maximum Energy Charge which I find useless and difficult to understand as it deals with the duration of a timestep, which is not explicit in the dataformat….. My suggestion was to keep Maximum Charge and Maximum Discharge but to remove Maximum Energy Charge
> 
> 4/ About the spillage. Yeah, I also agree. I can modify it.  Good Do you think that all these changes should be in the same PR?  Yes, won’t be too big…..
> 
> @erikfilias I think we can start the PR, right?

I'll modify it in #196 .
